examples/mutag_gin_equib.py

```diff
@@ -0,0 +1,100 @@
+import os.path as osp
+
+import torch
+import torch.nn.functional as F
+from torch.nn import BatchNorm1d, Linear, ReLU, Sequential
+
+from torch_geometric.datasets import TUDataset
+from torch_geometric.loader import DataLoader
+from torch_geometric.nn import EquilibriumAggregation, GINConv
+
+path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'TU')
+dataset = TUDataset(path, name='MUTAG')
+
+train_dataset = dataset[len(dataset) // 10:]
+test_dataset = dataset[:len(dataset) // 10]
+
+train_loader = DataLoader(train_dataset, batch_size=1)
+test_loader = DataLoader(test_dataset, batch_size=1)
+
+
+class Net(torch.nn.Module):
+    def __init__(self, in_channels, dim, out_channels):
+        super().__init__()
+
+        self.conv1 = GINConv(
+            Sequential(Linear(in_channels, dim), BatchNorm1d(dim), ReLU(),
+                       Linear(dim, dim), ReLU()))
+
+        self.conv2 = GINConv(
+            Sequential(Linear(dim, dim), BatchNorm1d(dim), ReLU(),
+                       Linear(dim, dim), ReLU()))
+
+        self.conv3 = GINConv(
+            Sequential(Linear(dim, dim), BatchNorm1d(dim), ReLU(),
+                       Linear(dim, dim), ReLU()))
+
+        self.conv4 = GINConv(
+            Sequential(Linear(dim, dim), BatchNorm1d(dim), ReLU(),
+                       Linear(dim, dim), ReLU()))
+
+        self.conv5 = GINConv(
+            Sequential(Linear(dim, dim), BatchNorm1d(dim), ReLU(),
+                       Linear(dim, dim), ReLU()))
+
+        self.lin1 = Linear(dim, dim)
+        self.lin2 = Linear(dim, out_channels)
+        self.readout = EquilibriumAggregation(dim, dim, [256, 256],
+                                              grad_iter=10, alpha=0.05)
+
+    def forward(self, x, edge_index, batch):
+        x = self.conv1(x, edge_index)
+        x = self.conv2(x, edge_index)
+        x = self.conv3(x, edge_index)
+        x = self.conv4(x, edge_index)
+        x = self.conv5(x, edge_index)
+        x = self.readout(x).unsqueeze(0)
```

Comment:
```suggestion
        x = self.readout(x)
```
Why do we need the `unsqueeze` here?

Reply:
right now the output shape of self.readout(x) is `(input_size)` but needs to be `(batch_size, input_size)` - the disparity comes because the model here is using batch but I've currently ignored that by setting batch to 1 because I wanted to simplify it as a I figured it out